service/cziti/sdk.go

```diff
@@ -517,20 +554,64 @@ func eventCB(ztx C.ziti_context, event *C.ziti_event_t) {
 				servicesToAdd = append(servicesToAdd, svcToAdd)
 			}
 		}
+		if len(servicesToAdd) > 0 {
+			for _, svc := range servicesToAdd {
+				if svc.Timeout >= 0 {
+					if minimumTimeout == -1 || minimumTimeout > svc.Timeout {
+						minimumTimeout = svc.Timeout
+					}
+					if allServicesTimeout == -1 || allServicesTimeout < svc.Timeout {
+						allServicesTimeout = svc.Timeout
+					}
+				}
+			}
+		}
+
+		zid.MinTimeout = minimumTimeout
+		zid.MaxTimeout = allServicesTimeout
+		zid.LastUpdatedTime = time.Now()
 
 		svcChange := BulkServiceChange{
 			Fingerprint:       zid.Fingerprint,
 			HostnamesToAdd:    hostnamesToAdd,
 			HostnamesToRemove: hostnamesToRemove,
 			ServicesToAdd:     servicesToAdd,
 			ServicesToRemove:  servicesToRemove,
+			MinTimeout:        minimumTimeout,
+			MaxTimeout:        allServicesTimeout,
+			LastUpdatedTime:   time.Now(),
 		}
 
 		if len(BulkServiceChanges) == cap(BulkServiceChanges) {
 			log.Warn("Service changes are not being processed fast enough. This client is out of date from the controller! This is unexpected. If you see this warning please report")
 		} else {
 			BulkServiceChanges <- svcChange
 		}
+	case C.ZitiMfaAuthEvent:
+		zid := (*ZIdentity)(appCtx)
+		log.Debugf("mfa auth event for finger print %s", zid.Fingerprint)
+		zid.MfaNeeded = true
+		zid.MfaEnabled = true
+
+		if zid.Fingerprint != "" {
+			var id = dto.Identity{
+				Name:              zid.Name,
+				FingerPrint:       zid.Fingerprint,
+				Active:            zid.Active,
+				ControllerVersion: zid.Version,
+				Status:            "",
+				MfaNeeded:         true,
+				MfaEnabled:        true,
+				Tags:              nil,
+			}
+
+			var m = dto.IdentityEvent{
+				ActionEvent: dto.IDENTITY_ADDED,
+				Id:          id,
+			}
+			goapi.BroadcastEvent(m)
+		}
+		log.Debugf("mfa auth event set enabled/needed to true for ziti context [%p]. Identity name:%s [fingerprint: %s]", zid, zid.Name, zid.Fingerprint)
```

Comment:
this debug is outside the if from line 596. maybe it should be inside the if - and an 'else' log message added when the fingerprint is empty? (which obviously shouldn't ever happen but - suppose 'might')